cpp/arcticdb/util/preconditions.hpp

```diff
@@ -21,7 +21,7 @@ struct Raise {
 
     template<typename...Args>
     [[noreturn]] void operator()(fmt::format_string<Args...> format, Args&&...args) const {
-        std::string combo_format = fmt::format("{} {}", error_code_data<code>.name_, format);
+        std::string combo_format = fmt::format("{} {}:{} {}", static_cast<std::underlying_type_t<ErrorCode>>(code), get_error_category_names().at(get_error_category(code)), error_code_data<code>.name_, format);
```

Comment:
The agreed format that I know of is `Ennnn: message`.

The idea is:
* The category name is encoded in the Exception name, which Python prints by default
* The `E_XXXX` mnemonic is for programmatic use only
so neither should appear in the error string

Is there a design change that I am not aware of?